core/state/statedb.go

```diff
@@ -368,6 +368,30 @@ func (s *StateDB) HasSelfDestructed(addr common.Address) bool {
 	return false
 }
 
+// CheckTransactionConditional validates the account preconditions against the statedb.
+func (s *StateDB) CheckTransactionConditional(cond *types.TransactionConditional) error {
+	for addr, acct := range cond.KnownAccounts {
```

Comment:
It would be nice enforce some sanity limit on the number of conditional accounts / storage dependencies. E.g. if there's more than 100, don't start any work, and just return an error. To avoid accidental DoS.

I know the surface RPC enforces some limit, but conditional txs still come in through p2p and perhaps other routes. Good to have a check on the `cond.Cost()` where it most matters.

Reply:
Sounds good, made the cost a public var in `types` that's also checked here